Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -225,9 +230,43 @@ public void run(@NotNull ProgressIndicator indicator) {
     @Override
     public void beforeDocumentChange(DocumentEvent e) {
       myServerData.onDocumentChanged(e);
+      synchronized (myLock) {
+        final Document document = e.getDocument();
+        myUpdatedDocuments.add(document);
+        scheduleUpdateFilesContent();
+      }
     }
   };
 
+  private void scheduleUpdateFilesContent() {
```

Comment:
Ah... And in this cause using maps of `DartServerData` will not work - we would not have subscriptions for invisible files :-(